### ISSUE TITLE
`api/v1/repos/:owner/:repo/pulls/***` not implemented

### ISSUE THREAD

**Issue by bkcsoft** [go-gitea/gitea#3472]
- Gitea version (or commit ref): ffd0062438a07dff73c949a0a963e62c5a29abf8
- Git version: N/A
- Operating system: N/A
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL): https://try.gitea.io/api/v1/repos/gitea/gitea/pulls/5/comments/
  - [ ] No
  - [ ] Not relevant

Seems like we don't have `/pulls`-endpoints implemented at all. I do remember them being implemented before though so they must've been removed in a refactor some time ago.

The following outputs are both for the same PR `#5`
```sh
$ curl https://try.gitea.io/api/v1/repos/gitea/gitea/issues/5/comments/ | jq .
[
  {
    "id": 2059,
    "html_url": "https://try.gitea.io/gitea/gitea/pulls/5#issuecomment-2059",
    "pull_request_url": "https://try.gitea.io/gitea/gitea/pulls/5",
    "issue_url": "",
    "user": {
      "id": 4,
      "login": "bkc",
      "full_name": "Kim \"BKC\" Carlbäcker",
      "email": "kim.carlbacker@gmail.com",
      "avatar_url": "https://secure.gravatar.com/avatar/d05dbfb80d6b9a461cd4578f18709a83?d=identicon",
      "username": "bkc"
    },
    "body": "Foobar",
    "created_at": "2018-02-08T01:32:01Z",
    "updated_at": "2018-02-08T01:32:01Z"
  }
]
```
```sh
$ curl -v https://try.gitea.io/api/v1/repos/gitea/gitea/pulls/5/comments/
[...]
> GET /api/v1/repos/gitea/gitea/pulls/5/comments/ HTTP/2
> Host: try.gitea.io
> User-Agent: curl/7.58.0
> Accept: */*
> 
* Connection state changed (MAX_CONCURRENT_STREAMS updated)!
< HTTP/2 404 
< content-type: text/plain; charset=utf-8
< date: Thu, 08 Feb 2018 01:35:09 GMT
< set-cookie: lang=en-US; Path=/; Max-Age=2147483647
< set-cookie: i_like_gitea=f71a34e0b91ce9db; Path=/; HttpOnly
< set-cookie: _csrf=RSpu7TlGgDC_5rh6Fw5QyPEkcUI6MTUxODA1MzcwOTExNDM2MzQwNg%3D%3D; Path=/; Expires=Fri, 09 Feb 2018 01:35:09 GMT; HttpOnly
< x-frame-options: SAMEORIGIN
< content-length: 0
< 
```

**Comment by Dunky13** [comment]
Looking at the api folder: `routers/api/v1`/repo/` you see the files `issue.go` & `issue_comment.go`

I'd expect the same for pull: `pull.go` & `pull_comment.go`

But the latter doesn't exist, and when searching through the history using:

`git log --stat --all --full-history -- routers/api/v1/repo/pull.*` it only shows `pull.go` as the only edited file.

If the `pull`-endpoints were implemented, then they weren't implemented the same way as `issue`-endpoints.

**Comment by bkcsoft** [comment]
@Dunky13 Commenting on PRs was just using `issue.Comment` IIRC. The only special code was in `pull.go`

**Comment by aliechti** [comment]
Is there any progress on this?

**Comment by lunny** [comment]
@thejahweh there is not still

**Comment by cljk** [comment]
> It's causing importing to GitLab to fail https://gitlab.com/gitlab-org/gitlab-ce/issues/31480#note_58059568

Can confirm it on my install - since I tried a migration of some projects to gitlab.

```

[Macaron] 2018-10-22 14:25:23: Completed GET /api/v1/repos/xxREDACTEDxx/xxREDACTEDxx/pulls/comments?page=1&per_page=100 404 Not Found in 3.914201ms
```
`gitea:1.6`

**Comment by debMan** [comment]
Importing from gitea to GitLab has some problems which I've faced and listed here:
1. API pagination on **issue** importing problem, which described in detail at  #9165. 
I've fixed this in wrong way, changed per page rows from 10 to 1000 on `modules/setting/setting.go` from `IssuePagingNum:      10,` to `IssuePagingNum:      1000,` and re-compiled. The issue importing process done.
2. Importing **pull requests** to GitLab which I'm working on
